include/nbl/builtin/hlsl/algorithm.hlsl

```diff
@@ -111,18 +109,27 @@ uint upper_bound(inout Accessor accessor, const uint begin, const uint end, cons
 namespace impl
 {
 
+template<typename T>
+struct comparator_lt_t
+{
+    bool operator()(const T lhs, const T rhs)
+    {
+        return lhs<rhs;
+    }
+};
+
```

Comment:
https://en.cppreference.com/w/cpp/utility/functional/less

define it in `nbl/builtin/hlsl/functional.hlsl`

keep to convention, call this `nbl::hlsl::less` surround in `#ifndef __HLSL_VERSION` and the C++ part do as `template<class T> using less = std::less;`